enum class MyEnum(val myArg: Int) {
    <!ENUM_USES_DEPRECATED_CONSTRUCTORS!>FIRST<!>: MyEnum(1),
    <!ENUM_USES_DEPRECATED_CONSTRUCTORS!>SECOND<!>: MyEnum(2),
    <!ENUM_USES_DEPRECATED_CONSTRUCTORS!>THIRD<!>: MyEnum(3),
    <!ENUM_USES_DEPRECATED_CONSTRUCTORS!>FOURTH<!>: MyEnum(4)
}